src/app/Rules/Rules.tsx

```diff
@@ -236,7 +289,20 @@ export const Rules = () => {
           <ToolbarContent>
             <ToolbarGroup variant="icon-button-group">
               <ToolbarItem>
-                <Button key="create" variant="primary" onClick={handleCreateRule}>Create</Button>
+                <Dropdown 
+                  onSelect= {()=>setIsDropdownOpen(false)} 
+                  toggle={
+                    <DropdownToggle
+                      toggleIndicator={null}
+                      onToggle={onRuleGenerationMenuToggle} 
```

Comment:
I think this could just be `onToggle={setIsDropdownOpen}` and the `React.useCallback` eliminated